_posts/2021-01-05-git-rebase-merge.md

```diff
@@ -0,0 +1,167 @@
+  auto: 0050-git
+---
+
+In this article we are going to discuss some very important commands from [git](https://git-scm.com/) and [GitHub](https://github.com), and how they make the life of developers easy - working individually or in a team. We will compare Git `rebase` with Git `merge` and explore some ways of using them in our Git workflow. We will start from the very basic and level up with each section.
+
+If you are a beginner to Git or GitHub and are looking to understand some of the basic commands used with git like fork, pull, checkout and commit then you should definitely give this [amazing article](https://reflectoring.io/github-fork-and-pull/) a read.
+
+## Introduction to Git
+
+Git is an **open-source distributed version control system**. We can break that work down into the following pieces:
+
+* **Control System:** Git can be used to store content – it is usually used to store code, but other content can also be stored.
+* **Version Control System:** Git helps in maintaining a history of changes and supports working on the same files in parallel by providing features like branching and merging.
+* **Distributed Version Control System:** The code is present in two types of repositories – the local repository, and the remote repository.
+
+## What is Git Merge?
+
+Now it's time to have a look at Git `merge`. A `merge` is a way to put a forked history back together. The git `merge` command lets us take independent branches of development and combine them into a single branch.
+
+It's important to note that while using the Git `merge`, the current branch will be updated to reflect the merge, but the target branch remains untouched. 
+
+`git merge` is often used in combination with `git checkout` for the selection of the current branch, and `git branch -d` for deleting the obsolete target branch.
+
+`git merge` is used for combining the multiple sequences of commits into one unified history. In the most common cases, we use `git merge` to combine two branches. 
+
+Let's take an example in which we will mainly focus on branch merging patterns. In the scenario which we have taken, `git merge` takes two commit pointers, and tries to find the common base commit between them. 
+
+Once Git has found a common base commit, it will create a new "merge commit", that will combine the changes of each queued merge commit sequence.
+
+![Git Merge combining branches](/assets/img/posts/git-rebase-merge/git-merge-working1.png)
+
+![Git Merge with Common Base](/assets/img/posts/git-rebase-merge/git-merge-working2.png)
+
+
+## What is Git Rebase?
+
+Let's have a look at the concept of Git `rebase`. A `rebase` is the way of migrating or combining a sequence of commits to a new base commit. If we consider it in the context of feature branching workflow, then it is more useful and easily visualized. The figure given below describes the general process of rebasing with Git:
+
+![Git Rebasing](/assets/img/posts/git-rebase-merge/git-rebasing-basic.png)
```

Comment:
I modified this image, have a look if it's ok.

Reply:
looks good